pkg/proxy/endpointslicecache.go

```diff
@@ -168,7 +182,20 @@ func (cache *EndpointSliceCache) getEndpointsMap(serviceNN types.NamespacedName,
 func (cache *EndpointSliceCache) endpointInfoByServicePort(serviceNN types.NamespacedName, sliceDataByName endpointSliceDataByName) spToEndpointMap {
 	endpointInfoBySP := spToEndpointMap{}
 
+	// Deterministic iteration order
+	sliceDataList := make([]*endpointSliceData, 0, len(sliceDataByName))
 	for _, sliceData := range sliceDataByName {
+		sliceDataList = append(sliceDataList, sliceData)
+	}
+	// Sort by trigger time and sequence
+	sort.Slice(sliceDataList, func(i, j int) bool {
```

Comment:
While the existing code is not particularly optimized, it's worth considering that 99.9% of the time this sorting/copying will be unnecessary. It might be better to just put off the extra comparisions until we need them (by splitting out a `preferredEndpoint()` function or whatever that gets called below only when we realize we have a duplicate).